### PR TITLE
Canvas: Enable ability to instantly select and drag an element

### LINKED ISSUE #52355: Canvas: Enable ability to instantly select and drag an element
It would be great if we could select and be able to drag an element right away on initial mousedown event. This functionality is currently commented out due to an issue where the `dragStart` event would be applied even after the user stopped clicking down on the element, causing the element to continue to follow the user's cursor

See #52343 for more context

Possible fixes may be refactoring our moveable / selecto instances to use React instead (maybe? given the demo seems to work)

Figuring out a way to prevent dragStart event from firing if the mouse is currently not actively being pressed down

### PR DESCRIPTION
**Pull request by drew08t**
What this PR does / why we need it:
When clicking an element that belongs to a frame, it jumps down, which can cause inconsistent dragging behavior.

Before:
Jul-27-2022 03-25-32

After:
Jul-27-2022 03-22-02

Fixes #52355

### CODE REVIEW

File: public/app/features/canvas/runtime/scene.tsx [hunk 1]
```diff
@@ -431,12 +430,12 @@ export class Scene {
       // @TODO Figure out click-drag functionality without phantom mouseup issue
       // https://github.com/daybrush/moveable/issues/481
```
Comment (nmarrs) [inline public/app/features/canvas/runtime/scene.tsx:432]:
```suggestion
```

File: public/app/features/canvas/runtime/scene.tsx [hunk 2]
```diff
@@ -431,12 +430,12 @@ export class Scene {
...
-      //   setTimeout(() => {
-      //     this.moveable!.dragStart(event.inputEvent);
-      //   });
-      // }
+      if (event.isDragStart) {
+        event.inputEvent.preventDefault();
+        setTimeout(() => {
+          this.moveable!.dragStart(event.inputEvent);
+        });
+      }
```
Comment (nmarrs) [inline public/app/features/canvas/runtime/scene.tsx:438]: Strange this is "just working" now when it wasn't before (due to phantom mouse down issue), but we'll take it haha

Just a few minor things to clean up:

1. when clicking and dragging the cursor doesn't update to be "grabbing" element

https://user-images.githubusercontent.com/22381771/181291090-79e62867-7abf-44bf-922e-974090c59c69.mov


2. MINOR - cursor behavior when dragging element underneath is a little wonky

https://user-images.githubusercontent.com/22381771/181291160-9f2aa8f5-3236-471c-b3d7-e18da0043740.mov

**Review by nmarrs — COMMENTED** [review]
Great job addressing issue with frame placement - I think we may want to change title / focus of PR more along the lines of ability to instantly click and drag an element though as the headliner :) 

The frame placement issue is good as a mention of pork (additional changes / fixes also taking a ride with the PR)

File: public/app/features/canvas/runtime/scene.tsx (hunk 1 above)
Comment (drew08t) [reply public/app/features/canvas/runtime/scene.tsx:432]: Good catch!

File: public/app/features/canvas/runtime/scene.tsx (hunk 2 above)
Comment (drew08t) [reply public/app/features/canvas/runtime/scene.tsx:438]: For part 2, I need to turn off pointer events for all other elements while dragging, then turn them back on when dragging is done. Need to find an efficient way to apply style to all OTHER elements.

**Comment by adela-almasan** [comment]
Just making note that the phantom mouse down is still happening for me locally. In the video, I'm just clicking on the element and then move the mouse and the element follows it. It doesn't happen all the time though.
While screen sharing with Drew, we could consistently reproduce when waiting for 5 seconds after refresh. 
Since it doesn't happen to anybody else, it might be something local on my side.

https://user-images.githubusercontent.com/88068998/181611366-d3d67bbc-5f8b-40ec-82b7-f35c62d8efa6.mov

File: public/app/features/canvas/runtime/element.tsx [hunk 3]
```diff
@@ -71,7 +71,7 @@ export class ElementState implements LayerElement {
   }
 
   /** Use the configured options to update CSS style properties directly on the wrapper div **/
-  applyLayoutStylesToDiv() {
+  applyLayoutStylesToDiv(disable?: boolean) {
```
Comment (nmarrs) [inline public/app/features/canvas/runtime/element.tsx:74]: Maybe we can be a bit more specific with this parameter as `disable` is pretty broad, maybe `disablePointerEvents`? (across all uses of this (including in the `scene` file))

File: public/app/features/canvas/runtime/scene.tsx [hunk 4]
```diff
@@ -332,6 +347,7 @@ export class Scene {
       })
       .on('dragStart', (event) => {
         this.ignoreDataUpdate = true;
+        this.disableNonTargets(event.target, true);
```
Comment (nmarrs) [inline public/app/features/canvas/runtime/scene.tsx:350]: I think this solution is ok given we are only firing this once on the beginning of drag event vs for every drag event 👍

**Review by nmarrs — COMMENTED** [review]
This is working really smoothly now! Just a nit on parameter naming but otherwise I think we are close to merge!

It may be a good idea to have 1-2 others (@ryantxu?) test this PR out locally just to verify if Adela's issue is something going on locally vs. potentially more widespread

**Review by nmarrs — APPROVED** [review]

**Pull request merged**